docs/architecture-decisions/2024-01-23_init-wizard-v2.md

```diff
@@ -0,0 +1,211 @@
+```
+
+**Enhancements to `.algokit.toml` enable the cli to:**
+
+- Navigate and interpret any project structure compliant with `.algokit.toml`.
+- Simplify interactions with project-specific commands.
+- Dynamically orchestrate command execution based on `.algokit.toml` configurations.
+  - For `algokit project {command_name}` executed at the workspace root, it aggregates and executes `{command_name}` from all child components either `sequentially` or `concurrently`, allowing users to choose the execution order per matching command identifier.
+  - When executed at a component's root, it runs the component-specific command from its `.algokit.toml`, also integrating it into the `algokit project` command group for visibility.
+
+**Suggested Improvements:**
+
+- Consider introducing command dependencies to ensure execution order for any command (in addition to current proposal to allow handling execution order for commands with matching 'names/identifiers'), enhancing the orchestration capabilities of the CLI.
+
+## Reference: Alternatives Considered
+
+### Consolidating AlgoKit Templates into a Monorepo
+
+**Overview:** Explores the possibility of merging AlgoKit templates into a single repository, focusing on streamlined project management and orchestration within js-centric environments.
+
+**Pros:**
+
+- Simplifies monorepo project management and orchestration in npm-based projects.
+
+**Cons:**
+
+- Increases complexity and potential conflicts with existing CLI capabilities and Copier's recommendations, suggesting that a better alternative might be to consider a shift to where we have a more CLI-centric approach that handles orchestration without introducing extra dependant pre-requisites.
+
+## Preferred Approach
+
+A phased implementation of both proposals, starting with user-facing enhancements to the `init` wizard followed by template and CLI command orchestration improvements. This strategy allows for immediate impact and addresses urgent needs first, with potential for further simplification through subsequent enhancements.
+
+## Next Steps
+
+1. Break down the selected options into detailed work items.
+2. Implement the proposed changes in the AlgoKit CLI and templates.
+3. Update documentation and onboarding materials to reflect the new changes.
+4. Announce the updates to the community and DevRel, soliciting feedback for future enhancements.
+5. Tackle minor improvements that become feasible following the main proposal implementations.
+
+### Open Questions
+
+1. Should we implement a bidirectional query mechanism where selecting a smart contract prompts the option to add a frontend, and similarly, choosing a frontend first triggers a query about integrating a smart contract? This approach necessitates a minor modification in the wording of the initial query. In the current workflow, we don't prompt adding smart contracts part since this question is prominently positioned as the initial query in the wizard's sequence.
```

Comment:
Also had this thought when reading the flowchart.

Reply:
Its a simple change, let's see on initial feedback once this is a bit more polished with tests and docs and we can add bidirectional flow here as needed